packages/api/src/platforms/vtex/resolvers/offer.ts

```diff
@@ -1,26 +1,116 @@
-import type { EnhancedSku } from '../utils/enhanceSku'
 import type { Resolver } from '..'
-import type { Item } from '../clients/commerce/types/Simulation'
+import type { StoreAggregateOffer } from './aggregateOffer'
+import {
+  getFirstSeller,
+  inStock,
+  inStockOrderFormItem,
+} from '../utils/productStock'
+import { getItemPriceByKey } from '../utils/price'
+import type { ArrayElementType } from '../../../typings'
+import type { Item } from '../clients/search/types/ProductSearchResult'
+import type { EnhancedSku } from '../utils/enhanceSku'
 import type { OrderFormItem } from '../clients/commerce/types/OrderForm'
 
-type Root =
-  | (Item & { product: EnhancedSku }) // when querying search/product
-  | (OrderFormItem & { product: Promise<EnhancedSku> }) // when querying order
+type OrderFormProduct = OrderFormItem & { product: Promise<EnhancedSku> }
+type SearchProduct = ArrayElementType<
+  ReturnType<typeof StoreAggregateOffer.offers>
+>
+type Root = SearchProduct | OrderFormProduct
+
+const isSearchItem = (item: any): item is Item => 'sellers' in item
+const isOrderFormItem = (item: any): item is OrderFormProduct =>
+  'skuName' in item
+
+const getAvailability = (available: boolean) =>
+  available ? 'https://schema.org/InStock' : 'https://schema.org/OutOfStock'
 
 export const StoreOffer: Record<string, Resolver<Root>> = {
   priceCurrency: () => '',
-  priceValidUntil: ({ priceValidUntil }) => priceValidUntil ?? '',
+  priceValidUntil: (item) => {
+    if (isSearchItem(item)) {
+      return getFirstSeller(item.sellers)?.commertialOffer.PriceValidUntil ?? ''
+    }
+
+    if (isOrderFormItem(item)) {
+      return item.priceValidUntil ?? ''
+    }
+
+    return null
+  },
   itemCondition: () => 'https://schema.org/NewCondition',
-  availability: ({ availability }) =>
-    availability === 'available'
-      ? 'https://schema.org/InStock'
-      : 'https://schema.org/OutOfStock',
-  seller: ({ seller }) => ({
-    identifier: seller,
-  }),
-  price: ({ sellingPrice }) => sellingPrice / 1e2, // TODO add spot price calculation
-  sellingPrice: ({ sellingPrice }) => sellingPrice / 1e2,
-  listPrice: ({ listPrice }) => listPrice / 1e2,
+  availability: async (item) => {
+    if (isSearchItem(item)) {
+      return getAvailability(!!inStock(item))
+    }
+
+    if (isOrderFormItem(item)) {
+      return getAvailability(inStockOrderFormItem(item.availability))
+    }
+
+    return null
+  },
+  seller: (item) => {
+    if (isSearchItem(item)) {
+      return {
+        identifier: getFirstSeller(item.sellers)?.sellerId ?? '',
+      }
+    }
+
+    if (isOrderFormItem(item)) {
+      return {
+        identifier: item.seller,
+      }
+    }
+
+    return null
+  },
+  price: (item) => {
+    if (isSearchItem(item)) {
+      return getItemPriceByKey(item, 'spotPrice') // TODO add spot price calculation
```

Comment:
```suggestion
      // TODO: add spot price calculation
      return getItemPriceByKey(item, 'spotPrice')
```

Reply:
You mean remove the TODO, right?

Reply:
nop, just realocate in the line above

Reply:
Do you think this TODO is necessary, since we're using the spotPrice now?